runtime(vim): Update base-syntax, improve script-interface command highlighting

dkearns opened the pull request:
Update base-syntax, improve script-interface command highlighting

- Normalise interface heredoc highlighting with that used for `:let-heredocs`.
- Remove interface feature testing.  The Lua and Python interface command scripts are now highlighted by default.  Loading all syntax files incurs an undesirable load-time burden so highlighting of the less popular MzScheme, Perl, Ruby and Tcl interfaces is disabled by default.  `g:vimsyn_embed` can still be used to customise the supported interfaces.
- Always highlight interface ex-commands as valid commands, even when the corresponding command-script highlighting is disabled.
- Remove error highlighting of heredoc and statement command-script regions when an interface is disabled.  These are now highlighted as plain text.
- Highlight simple command-script statement strings as well as heredocs.
- Allow indented heredoc end tokens when `trim` is specified.
- Match interface heredocs in `:def` functions.
- Fix runaway `vimEmbedError` regions.  These regions have been removed.
- Use python2 syntax for `:python`, and `:pythonx` when `'pyxversion'` is appropriately set.

k-takata commented on runtime/syntax/generator/vim.vim.base:
```diff
@@ -862,218 +863,266 @@ syn region	vimHiLink	contained matchgroup=Type start="\%(\<hi\%[ghlight]!\=\s\+\
+"   g:vimsyn_embed =~# 'p' : embed Perl
+"   g:vimsyn_embed =~# 'P' : embed Python
+"   g:vimsyn_embed =~# 'r' : embed Ruby
+"   g:vimsyn_embed =~# 't' : embed Tcl
```
I'm wondering if we need to check if vim support the script.
I sometimes want to enable the highlight even if the current vim doesn't support it.

dkearns replied:
Yes, I like to highlight all script commands all the time so I changed it to only highlight supported scripts by default but any user-set value of g:vimsyn_embed overrides that.  I think this behaviour matches the current documentation.

I don't think validating that a value in a user-set g:vimsyn_embed is actually supported is particularly useful.  Do you?

k-takata replied:
Ah, I misunderstood your change.
Looks good.

k-takata commented on runtime/syntax/generator/vim.vim.base:
```diff
@@ -862,218 +863,266 @@ syn region	vimHiLink	contained matchgroup=Type start="\%(\<hi\%[ghlight]!\=\s\+\
 else
- let s:trytcl= 1
+ let s:trytcl = 1
 endif
```
I'm wondering if we still need this part.
Trying to use Cygwin version of Tcl from Win32 version of Vim might cause hanging long time ago. But nowadays, Cygwin DLLs have the cyg prefix, so I don't think it can happen.

dkearns replied:
OK, I'll remove it.

dkearns committed:
runtime(vim): Update base-syntax, improve script-interface command highlighting

- Normalise interface heredoc highlighting with that used for
  :let-heredocs.
- Remove interface feature testing.  The Lua and Python interface
  command scripts are now highlighted by default.  Loading all syntax
  files incurs an undesirable load-time burden so highlighting of the
  less popular MzScheme, Perl, Ruby and Tcl interfaces is disabled by
  default.  g:vimsyn_embed can still be used to customise the supported
  interfaces.
- Always highlight interface ex-commands as valid commands, even when
  the corresponding command-script highlighting is disabled.
- Highlight simple command-script statements as well as heredocs.
- Remove error highlighting of heredoc and statement command-script
  regions when an interface is disabled.  These are now highlighted as
  plain text.
- Allow indented heredoc end tokens when "trim" is specified.
- Match interface heredocs in :def functions.
- Fix runaway vimEmbedError regions.  These regions have been removed.
- Use python2 syntax for :python, and :pythonx when 'pyxversion' is
  appropriately set.

Signed-off-by: Doug Kearns <dougkearns@gmail.com>
